shell/platform/darwin/ios/framework/Source/FlutterViewControllerTest.mm

```diff
@@ -430,6 +430,16 @@ - (void)testViewDidLoadDoesInvokeEngineWhenIsTheViewController {
   OCMVerify([mockEngine attachView]);
 }
 
+- (void)testSplashScreenViewRemoveNotCrash {
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"engine" project:nil];
+  [engine runWithEntrypoint:nil];
+  FlutterViewController* flutterViewController =
+      [[FlutterViewController alloc] initWithEngine:engine nibName:nil bundle:nil];
+  [flutterViewController setSplashScreenView:[[UIView alloc] init]];
+  UIView* nilView = nil;
+  [flutterViewController setSplashScreenView:nilView];
```

Comment:
@cyanglaz Emm, `setSplashScreenView:nil` would build error because `splashScreenView` declare `nonnull` and we treat build warning as error. Actually we have the case that let user pass the `nil` to remove the splash screen view, see https://github.com/flutter/engine/blob/257b05fda9bf30993d2b0cc11932c740c52386a3/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L630-L637.

Reply:
Isn't this wrong that we make the `view` paramter nonnull and still checking nullability in the implementation? Maybe we should update the method syntax to make the parameter nonnull.

Reply:
It looks like `setSplashScreenView` should allow nil #6834, probably an oversight from adding `NS_ASSUME_NONNULL_BEGIN`.  Can you instead make the API nullable?
```diff
- @property(strong, nonatomic) UIView* splashScreenView;
+ @property(strong, nonatomic, nullable) UIView* splashScreenView;
```
and add a comment to the header like:
>  `*` Set to `nil` to remove the splash screen view.

Reply:
@cyanglaz We support `view` nullable to remove the `view` from view hierarchy, but we use `NS_ASSUME_NONNULL_BEGIN` to declare `view` nonnull.

@jmagman We can make the API nullable, but it may lead to an API-breaking change for Swift users, if Swift users use the getter of `splashScreenView`, the return type changed from `UIView` to `UIView?`.

Reply:
https://github.com/flutter/engine/pull/34743 has merged, apologies for that taking so long.  Can you rebase on your change and make this `nil`?

Reply:
@jmagman Done. :)